io.c

```diff
@@ -9699,12 +9708,61 @@ static VALUE argf_readlines(int, VALUE *, VALUE);
 
 /*
  *  call-seq:
- *     readlines(sep=$/)     -> array
- *     readlines(limit)      -> array
- *     readlines(sep, limit) -> array
+ *    readlines(sep = $/)   -> array
+ *    readlines(limit)      -> array
+ *    readlines(sep, limit) -> array
  *
  *  Returns an array containing the lines returned by calling
- *  <code>Kernel.gets(<i>sep</i>)</code> until the end of file.
+ *  Kernel#gets until the end-of-file is reached.
+ *
+ *  With only string argument +sep+ given,
+ *  returns the remaining lines as determined by line separator +sep+,
+ *  or +nil+ if none;
+ *  see {Line Separator}[#class-IO-label-Line+Separator]:
+ *
+ *    f = File.new('t.txt')
+ *    f.readlines
+ *    # => ["First line\n", "Second line\n", "\n", "Fourth line\n", "Fifth line\n"]
+ *    f.rewind
+ *    f.readlines('li')
+ *    # => ["First li", "ne\nSecond li", "ne\n\nFourth li", "ne\nFifth li", "ne\n"]
+ *
+ *    File.write('try_readlines.rb', 'puts readlines')
+ *    `echo "test" | ruby try_readlines.rb` # => "\"test\" \n"
```

Comment:
I just understood we can do without `File.write` (here and in similar places) for test Ruby code; just do this: `-e` is flag to evaluate the code immediately
```suggestion
 *    `echo "test" | ruby -e "puts readlines"` # => "\"test\" \n"
```
Then we probably can drop the complexity of "run the system commands from inside the Ruby", and just do something like, it _should_ be self-explanative (or we can add "run this in your terminal"):

```
$ echo "test" | ruby -e "p readlines"
["test\n"]
```
(Also note I am using  `p`, it seems more demonstrative, preserving the shape of data: it is array of lines, and `\n` is included)